grpc/bcodes.go

```diff
@@ -52,13 +55,27 @@ func errorToCode(err error) codes.Code {
 		return NoSuchRegistrationError
 	case core.InternalServerError:
 		return InternalServerError
+	case *probs.ProblemDetails:
+		return ProblemDetails
 	default:
 		return codes.Unknown
 	}
 }
 
 func wrapError(err error) error {
-	return grpc.Errorf(errorToCode(err), err.Error())
+	code := errorToCode(err)
+	var body string
+	if code == ProblemDetails {
+		pd := err.(*probs.ProblemDetails)
+		bodyBytes, jsonErr := json.Marshal(pd)
+		if jsonErr != nil {
+			return err // fail open
```

Comment:
Can you comment that here, since to a casual read this looks wrong (every other error return explicitly wraps).